src/core/mount.c

```diff
@@ -1197,19 +1197,19 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         case MOUNT_MOUNTING_SIGKILL:
         case MOUNT_MOUNTING_SIGTERM:
 
-                if (f == MOUNT_SUCCESS)
-                        mount_enter_mounted(m, f);
+                if (m->result == MOUNT_SUCCESS)
+                        mount_enter_mounted(m, m->result);
```

Comment:
Hmm, maybe just remove these two lines? This is what caused your original problem right? Somewhere later in this function is a comment, that from_proc_self_mountinfo is up-to-date  when this is called, so the stuff below should do the right thing.